tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/CalDavEventAttendanceRepository.java

```diff
@@ -0,0 +1,182 @@
+import com.linagora.tmail.james.jmap.model.EventAttendanceStatusEntry;
+import com.linagora.tmail.james.jmap.model.LanguageLocation;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import scala.collection.JavaConverters;
+
+public class CalDavEventAttendanceRepository implements EventAttendanceRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CalDavEventAttendanceRepository.class);
+    private static final String X_MEETING_UID_HEADER = "X-MEETING-UID";
+
+    private final DavClient davClient;
+    private final SessionProvider sessionProvider;
+    private final MessageId.Factory messageIdFactory;
+    private final MessageIdManager messageIdManager;
+    private final DavUserProvider davUserProvider;
+
+    @Inject
+    public CalDavEventAttendanceRepository(DavClient davClient,
+                                           SessionProvider sessionProvider, MessageId.Factory messageIdFactory,
+                                           MessageIdManager messageIdManager,
+                                           DavUserProvider davUserProvider) {
+        this.davClient = davClient;
+        this.sessionProvider = sessionProvider;
+        this.messageIdFactory = messageIdFactory;
+        this.messageIdManager = messageIdManager;
+        this.davUserProvider = davUserProvider;
+    }
+
+
+    @Override
+    public Publisher<CalendarEventAttendanceResults> getAttendanceStatus(Username username,
+                                                                         BlobIds calendarEventBlobIds) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return Flux.fromIterable(JavaConverters.seqAsJavaList(calendarEventBlobIds.value()))
+            .flatMap(blobId -> getEventUid(blobId.value(), systemMailboxSession)
+                .flatMap(eventUid ->
+                    toDavUser(username)
+                        .flatMap(davUser -> davClient.getCalendarObjectContainingVEvent(davUser, eventUid)))
+                .map(calendarObject -> getAttendanceStatusFromCalendarObject(
+                    BlobId.of(blobId.value()).get(), calendarObject, username, systemMailboxSession)))
+            .reduce(CalendarEventAttendanceResults$.MODULE$.empty(), CalendarEventAttendanceResults$.MODULE$::merge);
+    }
+
+    private CalendarEventAttendanceResults getAttendanceStatusFromCalendarObject(BlobId blobId,
+        DavCalendarObject calendarObject, Username username, MailboxSession mailboxSession) {
+
+        try {
+            List<CalendarEventParsed> events = JavaConverters.asJava(CalendarEventParsed.from(calendarObject.calendarData()));
+
+            if (events.size() != 1) {
+                if (events.isEmpty()) {
+                    LOGGER.debug("""
+                No VEvents found in calendar object.
+                Returning empty attendance results.
+                """);
```

Comment:
Please kinly stop doing multiline logs

Reply:
Not used. This code is from an outdated commit.